hack/composefiles/compose.yaml

```diff
@@ -5,7 +5,8 @@ services:
     build:
       context: ../..
       args:
-        BUILDKIT_DEBUG: 1
+        #BUILDKIT_DEBUG: 1
+        CGO_ENABLED: 1
```

Comment:
For CGO mode-free you can look at https://pkg.go.dev/modernc.org/sqlite

Reply:
I think we can consider this if we need a CGO free mode. At the current moment, we still have the bolt database and we can just easily say "this feature only exists with cgo". I took a look at that project and the C library seems to be faster in most places.

I updated the PR to include some better messages for when cgo is disabled and to also only enable it on linux and darwin which shouldn't even matter much because I don't think you can run buildkit on darwin or windows anyway.